dotenv-linter/tests/common/test_dir.rs

```diff
@@ -14,20 +15,23 @@ use std::str::from_utf8;
 /// Use to test commands in temporary directories
 pub struct TestDir {
     current_dir: TempDir,
+    envs: Option<HashMap<String, String>>,
 }
 
 impl TestDir {
     /// Generate a new TestDir
-    pub fn new() -> Self {
+    pub fn new(envs: Option<HashMap<String, String>>) -> Self {
```

Comment:
You can don't change `new`, just encapsulate `None` there. And add one more constructor `with_envs`.

Second option is pass ENV variables directly to methods as optional argument, but it would involve all usages. So it could be also two- definition per method (to avoid changes in other places)

Reply:
Thank you for the pointers, they were really helpful! Updated here: https://github.com/dotenv-linter/dotenv-linter/pull/625/commits/7d85e99ffc5c79e4c52eb386e7875cf5d2aaa6a9